citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java

```diff
@@ -149,7 +150,25 @@ protected void handleExportRequest(HttpServletRequest req, HttpServletResponse r
 		
 		if(org.sakaiproject.citation.api.CitationService.RIS_FORMAT.equals(format))
 		{
-			String citationCollectionId = req.getParameter("citationCollectionId");
+			String citationCollectionId = null;
+			ContentResource resource = null;
+			try {
+				resource = ContentHostingService.getResource(req.getParameter("resourceId"));
+				citationCollectionId = new String(resource.getContent());
+			}
+			catch (PermissionException e) {
+				throw new EntityPermissionException(SessionManager.getCurrentSessionUserId(), "handleExportRequest", ref.getReference());
```

Comment:
It's much better to use the ComponentManager to get an instance of a service as the static covers should be going away.